agent-control/src/agent_type/runtime_config/version_config.rs

```diff
@@ -0,0 +1,19 @@
+use serde::Deserialize;
+
+use crate::agent_type::version_config::VersionCheckerInterval;
+
+/// Represents the configuration for version checks.
+#[derive(Debug, Deserialize, Clone, PartialEq)]
+pub struct OnHostVersionConfig {
+    /// The command to check the version.
+    #[serde(default)]
+    pub(crate) command: String,
+
+    /// The regex expression to get the version from the command output.
+    #[serde(default)]
+    pub(crate) regex: Option<String>,
+
+    /// The duration to wait between version checks.
+    #[serde(default)]
+    pub(crate) interval: VersionCheckerInterval,
```

Comment:
Not totally sure what to do with that.

If no one can think of a reason, I guess we should remove it and implement it back when a client complains.

Reply:
That's another option yeah.

Reply:
Wait, you mean leaving the interval option here and running it once when it's `None` for example? Or you mean removing it and just running it once?

Reply:
I mean , removing the extra thread and just run it once, if that makes sense

Reply:
@gsanchezgavier @sigilioso could it be that the reason to have a loop for the version checker is because in k8s, stuff doesn't have to exist at the beginning, but it's eventual consistent?

I ask because of a failing test `k8s_test_attributes_from_existing_agent_type`.